src/helpers/date-helpers.ts

```diff
@@ -1,6 +1,16 @@
-import { add, Duration, isSameDay, sub } from "date-fns";
+import { add, Duration, isSameDay, sub, Day, parseISO, formatRelative, formatDistanceToNow } from "date-fns";
 import language from "./language";
-import { formatDateForLocale } from "./locale";
+import { formatDateForLocale, getDateLocale, getIntlLocale, capitalizeFirstLetterForLocale } from "./locale";
+
+export function toDate(dateOrDateString: string | Date): Date {
+    var date;
+    if (typeof(dateOrDateString) === 'string') {
+        date = parseISO(dateOrDateString);
+    } else {
+        date = dateOrDateString;
+    }
+    return date;
```

Comment:
I echo the need to validate

Reply:
@reneefromhold `parseISO` already does validation on the date string. I don't think we gain anything by trying to second-guess it.

Reply:
@lynnfaraday I'm finding when passing an empty string to parseISO it returns Invalid Date, this creates an error downstream in formatDateForLocale

Reply:
OK. It seems that we want "Invalid Date" to fall through to the UI though rather than some random console error? That would require a check in formatDateForLocale

Reply:
That works. This is where it was blowing up [earlier](https://github.com/CareEvolution/MyDataHelpsUI/blob/fd076ff0ff45a2a712833c5fe476cf73f53ada66/src/helpers/locale.ts#L117)